### ISSUE TITLE
Table counter is overrided by code-block's in LaTeX

### ISSUE THREAD

**Issue by tk0miya** [sphinx-doc/sphinx#2319]
In Sphinx-1.3.5, the counter of tables is overrided by code-block's in latex output.
You can see the caption of table starts with 'Listings ...' in following screenshot:
![2016-02-14 14 06 36](https://cloud.githubusercontent.com/assets/748828/13031963/31eac3b4-d324-11e5-923e-dd2d0b2f08c0.png)

The screenshot above is generated from following reST:

```
.. list-table:: hello-table
   :header-rows: 1

   * - Hello
     - World
   * - Hello
     - World

.. code-block:: python
   :caption: hello-world

   print('Hello world')

.. code-block:: python
   :caption: hello-world

   print('Hello world')

.. list-table:: hello-table
   :header-rows: 1

   * - Hello
     - World
   * - Hello
     - World
```

This behavior comes from the combination of `threeparttable` and `captionof`.
As a test, I wraped the `threeparttable` in `table` environment. Then the problem was fixed.

**Comment by jfbu** [comment]
Screenshot if using https://github.com/jfbu/sphinx/commit/59de4bf71763cf20dfe86ce8c5185f99af4ea2b5 
(commit branch on top of 1.3.5 Sphinx release)

![capture d ecran 2016-02-14 a 10 08 24](https://cloud.githubusercontent.com/assets/2589111/13032844/52a27686-d303-11e5-9fe0-6419ddbae504.png)

**Comment by jfbu** [comment]
This does not change the fact that indeed `threeparttable` environment should better be used from inside a `table` environment with suitable caption therein.

**Comment by tk0miya** [comment]
@jfbu Sorry, I cannot determine this change is stable or not (because I'm not good at LaTeX macros...).
Do you know this change has any side effects? Does it have any limitation?

**Comment by tk0miya** [comment]
Fixed by #2297.
